source/parser.ts

```diff
@@ -86,3 +88,112 @@ export function parseTextTable<T = {}>(
                   }, {}) as T
           );
 }
+
+function isQuoteIncomplete(buffer: string): boolean {
+    // Check if buffer contains an unmatched opening quote
+    let inQuote = false;
+    let quote = '';
+
+    for (let i = 0; i < buffer.length; i++) {
+        const char = buffer[i];
+        if ((char === '"' || char === "'") && !inQuote) {
+            inQuote = true;
+            quote = char;
+        } else if (char === quote && inQuote) {
+            inQuote = false;
+            quote = '';
+        }
+    }
+
+    return inQuote;
+}
+
+export async function* readTextTable<T = {}>(
+    chunks: AsyncIterable<string>,
+    header?: boolean,
+    separator = ','
+): AsyncGenerator<T extends {} ? T : any[], void, unknown> {
+    let buffer = '';
+    let headerRow: any[] | undefined;
+    let isFirstRow = true;
+
+    for await (const chunk of chunks) {
+        buffer += chunk;
+
+        // Process buffer character by character to find complete rows
+        let rowStart = 0;
+        let inQuote = false;
+        let quoteChar = '';
+
+        for (let i = 0; i < buffer.length; i++) {
+            const char = buffer[i];
+
+            // Handle quote state
+            if ((char === '"' || char === "'") && !inQuote) {
+                inQuote = true;
+                quoteChar = char;
+            } else if (char === quoteChar && inQuote) {
+                inQuote = false;
+                quoteChar = '';
+            }
+
+            // Check for row ending (newline) when not in quotes
+            if (!inQuote && (char === '\n' || char === '\r')) {
+                const rowData = buffer.slice(rowStart, i).trim();
+                if (rowData) {
+                    const parsedRow = parseRow(rowData, separator);
+
+                    if (header && isFirstRow) {
+                        headerRow = parsedRow;
+                        isFirstRow = false;
+                    } else {
+                        if (header && headerRow) {
+                            const rowObject = parsedRow.reduce(
+                                (object, item, index) => {
+                                    object[headerRow![index]] = item;
+                                    return object;
+                                },
+                                {} as any
+                            );
+                            yield rowObject as T extends {} ? T : any[];
+                        } else {
+                            yield parsedRow as T extends {} ? T : any[];
+                        }
```

Comment:
```suggestion
        const makeObject = (parsedRow: string[]) => parsedRow.reduce(
                                (object, item, index) => {
                                    object[headerRow![index]] = item;
                                    return object;
                                },
                                {} as T
                            );

                        yield (header && headerRow) ?
                            makeObject(parsedRow) :
                            parsedRow;
```
下面同样的逻辑也这么改，用箭头函数复用核心逻辑。